tensorflow/core/kernels/sequence_ops.cc

```diff
@@ -82,27 +102,23 @@ class RangeOp : public OpKernel {
     OP_REQUIRES_OK(context, shape.AddDimWithStatus(size));
     Tensor* out = nullptr;
     OP_REQUIRES_OK(context, context->allocate_output(0, shape, &out));
+    if (size == 0) return;
     auto flat = out->flat<T>();
-    T val = start;
-    for (int64_t i = 0; i < size; ++i) {
-      flat(i) = T(val);
-      val += delta;
-    }
+    functor::RangeFunctor<Device, T>()(context, size, start, delta, flat);
   }
 };
 
-#define REGISTER_KERNEL(DEV, TYPE)                           \
+#define REGISTER_KERNEL(DEV, DEV_TYPE, TYPE)                 \
   REGISTER_KERNEL_BUILDER(Name("Range")                      \
                               .Device(DEV)                   \
                               .HostMemory("start")           \
                               .HostMemory("limit")           \
```

Comment:
Thank you for the contribution!

Currently there are many ops that require inputs and/or outputs in host memory for int32.
As a result, in the current state, it leads to performance degradation in our benchmarks in some int32 use-cases due to the extra D2H copies.

I suggest using the existing implementation for the int32 case, with output in host memory.

One way to achieve this would be to define an additional registration macro here (e.g. REGISTER_HOST_OUTPUT_KERNEL), use it in REGISTER_CPU_KERNEL below, and remove the 'TF_CALL_int32(REGISTER_GPU_KERNEL)' registration. 
The CPU implementation with output in host memory will then be used as the fallback for int32.

Reply:
Done